### PR TITLE
Implemented NSI validity checks

### PR DESCRIPTION
**Pull request by AndyHWChung**

### CODE REVIEW

File: app/uk/gov/hmrc/helptosavefrontend/models/NSIUserInfo.scala [hunk 1]
```diff
@@ -0,0 +1,290 @@
...
+      val lengthCheck =
+        validatedFromBoolean(p)(_.length <= 10, "Postcode was longer thn 10 characters")
+
+      val regexCheck = regexValidation(p)(postcodeRegex, "Invalid postcode format")
+
+      (lengthCheck |@| regexCheck).tupled.map(_ ⇒ p)
+  }
+
+  private def countryCodeValidation(countryCode: Option[String]): ValidatedNel[String, Option[String]] =
+    validatedFromBoolean(countryCode.map(_.trim))(_.forall(_.length == 2), "Country code was not ")
```
Comment (cavster) [inline app/uk/gov/hmrc/helptosavefrontend/models/NSIUserInfo.scala]: Country code was not two characters maybe

File: app/uk/gov/hmrc/helptosavefrontend/models/NSIUserInfo.scala [hunk 2]
```diff
@@ -0,0 +1,290 @@
...
+    }
+
+    (lengthCheck |@| lineCheck).map( (_, lines) ⇒ lines)
+  }
+
+  private def postcodeValidation(postcode: Option[String]): ValidatedNel[String,String] = postcode match{
+    case None ⇒
+      Invalid(NonEmptyList.of("Postcode undefined"))
+
+    case Some(p) ⇒
```
Comment (cavster) [inline app/uk/gov/hmrc/helptosavefrontend/models/NSIUserInfo.scala]: Do we want a check for post code not being enough characters?

File: app/uk/gov/hmrc/helptosavefrontend/models/NSIUserInfo.scala [hunk 3]
```diff
@@ -0,0 +1,290 @@
...
+
+      (lengthCheck |@| regexCheck).tupled.map(_ ⇒ p)
+  }
+
+  private def countryCodeValidation(countryCode: Option[String]): ValidatedNel[String, Option[String]] =
+    validatedFromBoolean(countryCode.map(_.trim))(_.forall(_.length == 2), "Country code was not ")
+
+  private def ninoValidation(nino: NINO): ValidatedNel[String, NINO] = {
+    val lengthCheck =
+      validatedFromBoolean(nino)(_.length <= 9, "NINO was longer thn 9 characters")
```
Comment (cavster) [inline app/uk/gov/hmrc/helptosavefrontend/models/NSIUserInfo.scala]: Same with Nino or does it matter?

File: app/uk/gov/hmrc/helptosavefrontend/models/NSIUserInfo.scala (hunk 1 above)
Comment (AndyHWChung) [inline app/uk/gov/hmrc/helptosavefrontend/models/NSIUserInfo.scala]: NSI want them to be of length 2

File: app/uk/gov/hmrc/helptosavefrontend/models/NSIUserInfo.scala (hunk 2 above)
Comment (AndyHWChung) [inline app/uk/gov/hmrc/helptosavefrontend/models/NSIUserInfo.scala]: the regex should pick this up

File: app/uk/gov/hmrc/helptosavefrontend/models/NSIUserInfo.scala (hunk 3 above)
Comment (AndyHWChung) [inline app/uk/gov/hmrc/helptosavefrontend/models/NSIUserInfo.scala]: again, the regex should pick this up

**Pull request merged**